src/lib/statistics/statistics_objects/min_max_filter.cpp

```diff
@@ -158,6 +159,40 @@ bool MinMaxFilter<T>::does_not_contain(const PredicateCondition predicate_condit
       const auto value2 = boost::get<T>(*variant_value2);
       return value >= max || value2 <= min;
     }
+    case PredicateCondition::Like: {
+      // Examples for the handle of Like predicate:
+      //                        | test%         | %test   | test\x7F% | test           | '' (empty string)
+      // LikeMatcher::bounds()  | {test, tesu}  | nullopt | nullopt   | {test, test\0} | {'', '\0'}
+      // does_not_contain(Like) | max < test || | false   | false     | max < test ||  | max < '' ||
+      //                        | min >= tesu   |         |           | min >= test\0  | min >= '\0'
+      if constexpr (std::is_same_v<T, pmr_string>) {
+        const auto bounds = LikeMatcher::bounds(value);
+        if (!bounds) return false;
+
+        const auto [lower_bound, upper_bound] = bounds.value();
+
+        return max < lower_bound || upper_bound <= min;
+      }
+
+      return false;
+    }
+    case PredicateCondition::NotLike: {
+      // Examples for the handle of NotLike predicate:
+      //                          | test%          | %test   | test\x7F% | test            | '' (empty string)
+      // LikeMatcher::bounds()    | {test, tesu}   | nullopt | nullopt   | {test, test\0}  | {'', '\0'}
+      // does_not_contain(NotLike)| min >= test && | false   | false     | min >= test &&  | min >= '\0' &&
+      //                          | max < tesu     |         |           | max < test\0    | max < '\0'
+      if constexpr (std::is_same_v<T, pmr_string>) {
+        const auto bounds = LikeMatcher::bounds(value);
+        if (!bounds) return false;
+
+        const auto [lower_bound, upper_bound] = bounds.value();
+
+        return max < upper_bound && lower_bound <= min;
+      }
+
+      return false;
```

Comment:
I ran CI with FullCI: clang-tidy has a false positive here when it considers the `else` path of the `if constexpr`. Feel free to supress or let me know if you need a hand. :)

Reply:
I think I would suppress it. For that, I could simply add `// NOLINT` to line 194 (with a small description) correct?

Reply:
Yepp

Reply:
You might have to play around with the line number. 162 is a candidate, too. There is also `// NOLINTNEXTLINE - blabla` which keeps the nolint and the comment together.